source/server/filter_chain_manager_impl.cc

```diff
@@ -25,13 +25,118 @@ Network::Address::InstanceConstSharedPtr fakeAddress() {
+  return parent_context_.direction();
+}
+
+ProtobufMessage::ValidationVisitor& FilterChainFactoryContextImpl::messageValidationVisitor() {
+  return parent_context_.messageValidationVisitor();
+}
+
+AccessLog::AccessLogManager& FilterChainFactoryContextImpl::accessLogManager() {
+  return parent_context_.accessLogManager();
+}
+
+Upstream::ClusterManager& FilterChainFactoryContextImpl::clusterManager() {
+  return parent_context_.clusterManager();
+}
+
+Event::Dispatcher& FilterChainFactoryContextImpl::dispatcher() {
+  return parent_context_.dispatcher();
+}
+
+Grpc::Context& FilterChainFactoryContextImpl::grpcContext() {
+  return parent_context_.grpcContext();
+}
+
+bool FilterChainFactoryContextImpl::healthCheckFailed() {
+  return parent_context_.healthCheckFailed();
+}
+
+Tracing::HttpTracer& FilterChainFactoryContextImpl::httpTracer() { return httpContext().tracer(); }
+
+Http::Context& FilterChainFactoryContextImpl::httpContext() {
+  return parent_context_.httpContext();
+}
+
+const LocalInfo::LocalInfo& FilterChainFactoryContextImpl::localInfo() const {
+  return parent_context_.localInfo();
+}
+
+Envoy::Runtime::RandomGenerator& FilterChainFactoryContextImpl::random() {
+  return parent_context_.random();
+}
+
+Envoy::Runtime::Loader& FilterChainFactoryContextImpl::runtime() {
+  return parent_context_.runtime();
+}
+
+Stats::Scope& FilterChainFactoryContextImpl::scope() { return parent_context_.scope(); }
+
+Singleton::Manager& FilterChainFactoryContextImpl::singletonManager() {
+  return parent_context_.singletonManager();
+}
+
+OverloadManager& FilterChainFactoryContextImpl::overloadManager() {
+  return parent_context_.overloadManager();
+}
+
+Admin& FilterChainFactoryContextImpl::admin() { return parent_context_.admin(); }
+
+TimeSource& FilterChainFactoryContextImpl::timeSource() { return api().timeSource(); }
+
+Api::Api& FilterChainFactoryContextImpl::api() { return parent_context_.api(); }
+
+ServerLifecycleNotifier& FilterChainFactoryContextImpl::lifecycleNotifier() {
+  return parent_context_.lifecycleNotifier();
+}
+
+OptProcessContextRef FilterChainFactoryContextImpl::processContext() {
+  return parent_context_.processContext();
+}
+
+Configuration::ServerFactoryContext&
+FilterChainFactoryContextImpl::getServerFactoryContext() const {
+  return parent_context_.getServerFactoryContext();
+}
+
+Stats::Scope& FilterChainFactoryContextImpl::listenerScope() {
+  return parent_context_.listenerScope();
+}
+
 bool FilterChainManagerImpl::isWildcardServerName(const std::string& name) {
   return absl::StartsWith(name, "*.");
 }
 
 void FilterChainManagerImpl::addFilterChain(
     absl::Span<const envoy::config::listener::v3alpha::FilterChain* const> filter_chain_span,
-    FilterChainFactoryBuilder& filter_chain_factory_builder) {
+    FilterChainFactoryBuilder& filter_chain_factory_builder,
+    FilterChainFactoryContextCreator& callback) {
```

Comment:
nit: can you s/callback/context_creator (or similar) here and elsewhere?

Reply:
done